internal/pkg/opentelemetry/controller.go

```diff
@@ -30,21 +30,35 @@ type Controller struct {
 	logger         logr.Logger
 	version        string
 	tracerProvider trace.TracerProvider
-	tracersMap     map[string]trace.Tracer
+	tracersMap     map[tracerID]trace.Tracer
 	bootTime       int64
 }
 
-func (c *Controller) getTracer(pkg string) trace.Tracer {
-	t, exists := c.tracersMap[pkg]
+type tracerID struct{ name, version, schema string }
+
+func (c *Controller) getTracer(pkg, tracerName, version, schema string) trace.Tracer {
+	tID := tracerID{name: pkg, version: c.version}
```

Comment:
Doesn't this need to include the schema?

```suggestion
	tID := tracerID{name: pkg, version: c.version, schema: schema}
```